### PR TITLE
feat: provide sample code for row-level error handling

### PR DESCRIPTION
**Pull request by agrawal-siddharth**

### CODE REVIEW

File: samples/snippets/src/main/java/com/example/bigquerystorage/WriteToDefaultStream.java [hunk 1]
```diff
@@ -191,7 +197,37 @@ public void onFailure(Throwable throwable) {
...
+              }
+            }
+
+            // Mark the existing attempt as done since we got a response for it
+            done();
+
+            // Retry the remaining valid rows.
+            if (dataNew.length() > 0) {
+              try {
+                this.parent.append(new AppendContext(dataNew, 0));
```
Comment (yirutang) [inline samples/snippets/src/main/java/com/example/bigquerystorage/WriteToDefaultStream.java:224]: Is there a way to start another thread and do this append? Currently this append can block waiting for inflight limit...

File: samples/snippets/src/test/java/com/example/bigquerystorage/WriteToDefaultStreamIT.java [hunk 2]
```diff
@@ -73,7 +73,11 @@ public void setUp() {
     // Create a new dataset and table for each test.
     datasetName = "WRITE_STREAM_TEST" + UUID.randomUUID().toString().substring(0, 8);
     tableName = "DEFAULT_STREAM_TEST" + UUID.randomUUID().toString().substring(0, 8);
-    Schema schema = Schema.of(Field.of("test_string", StandardSQLTypeName.STRING));
+    Schema schema = Schema.of(
```
Comment (yirutang) [inline samples/snippets/src/test/java/com/example/bigquerystorage/WriteToDefaultStreamIT.java:76]: Where is the actual test?

Comment (agrawal-siddharth) [reply samples/snippets/src/test/java/com/example/bigquerystorage/WriteToDefaultStreamIT.java:76]: https://github.com/googleapis/java-bigquerystorage/blob/main/samples/snippets/src/test/java/com/example/bigquerystorage/WriteToDefaultStreamIT.java

has the function testWriteToDefaultStream(). This calls WriteToDefaultStream.writeToDefaultStream() which does the test.

File: samples/snippets/src/main/java/com/example/bigquerystorage/WriteToDefaultStream.java (hunk 1 above)
Comment (agrawal-siddharth) [reply samples/snippets/src/main/java/com/example/bigquerystorage/WriteToDefaultStream.java]: Done. Both retries and row-errors are now handled by calls to append() made from a separate thread.

**Review by yirutang — APPROVED** [review]

**Pull request merged**